File: app/src/main/java/com/chooloo/www/callmanager/ui/base/BaseContract.java
package com.chooloo.www.callmanager.ui.base;

import android.content.Context;

import androidx.lifecycle.Lifecycle;

public interface BaseContract {
    interface View {
    }

    interface Presenter<V extends View> {
        void bind(V view, Lifecycle lifecycle);

        void unbind();

        void subscribe(Context context);
    }
}

File: app/src/main/java/com/chooloo/www/koler/ui/base/BaseContract.kt
package com.chooloo.www.koler.ui.base

import androidx.annotation.StringRes
import androidx.lifecycle.Lifecycle
import com.chooloo.www.koler.di.activitycomponent.ActivityComponent

interface BaseContract {
    interface View {
        val component: ActivityComponent

        fun onSetup()
        fun finish() {}
        fun getLifecycle(): Lifecycle
        fun showError(@StringRes stringResId: Int)
        fun showMessage(@StringRes stringResId: Int)
    }

    interface Controller<V : View> {
        val view: V
    }
}